File: ws-lib/src/main/java/com/tim/iot/ws/entity/WebSocketCloseEnum.java
package com.tim.iot.ws.entity;

/**
 * @author Tell.Tim
 * @date 2019/12/3 14:36
 */
public enum WebSocketCloseEnum {
    /**
     * websocket关闭枚举
     */
    USER_EXIT(1000,"close");
    private int code;
    private String reason;

    WebSocketCloseEnum(int code, String reason) {
        this.code = code;
        this.reason = reason;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }
}

File: app/src/main/java/com/tim/iot/auth/entity/WebSocketCloseEnum.kt
package com.tim.iot.auth.entity

/**
 * @author Tell.Tim
 * @date 2019/12/3 14:36
 */
enum class WebSocketCloseEnum private constructor(var code: Int, var reason: String?) {
    /**
     * websocket关闭枚举
     */
    USER_EXIT(1000, "close"),
    FORCE_EXIT(3000, "timeout close")
}
